src/layout.ts

```diff
@@ -63,7 +63,21 @@ export class Attribute {
      *        For type "FLOAT", this parameter has no effect.
      */
     constructor(public key: string, public size: number, public type: TYPES, public normalized: boolean = false) {
-        this.sizeOfType = this.type;
+        switch(type) {
+            case "BYTE":
+            case "UNSIGNED_BYTE":
+                this.sizeOfType = 1;
+                break;
+            case "SHORT":
+            case "UNSIGNED_SHORT":
+                this.sizeOfType = 2;
+                break;
+            case "FLOAT":
+                this.sizeOfType = 4;
+                break;
+            default:
+                throw new Error(`Unknown gl type: ${type}`);
```

Comment:
I have it throwing an Error here instead. Thoughts?

Reply:
I like that much better. Typically I prefer loud errors (especially when it's most likely the user author's fault) that halt rather than continuing on and hoping for the best.